pages/index.js

```diff
@@ -1144,9 +1145,17 @@ export async function getStaticProps() {
   const consoleCount = await getConsoles()
 
   // Hackathons: get latest hackathons
-  const hackathonsData = await fetch(
-    'https://hackathons.hackclub.com/api/events/upcoming'
-  ).then(res => res.json())
+  let hackathonsData;
+  try {
+    const response = await fetch('https://hackathons.hackclub.com/api/events/upcoming');
+    if (response.ok) {
+      hackathonsData = await response.json();
+    } else {
+      hackathonsData = []; // or some default value if the fetch fails
+    }
+  } catch (error) {
+    hackathonsData = []; // or some default value if an error occurs
```

Comment:
Just so I understand - the fallback here would be to show no hackathons as upcoming in the case where it couldn't hit the hackathons backend as opposed to throwing a 500 the way it does now?

Could it be argued that it might be better to throw a 500 and not lie?

Reply:
Maybe we log the error? The benefit of showing an empty array is that it completely hides the scrolling events instead of erroring:
https://hackclub.com/hackathons/
(I think right now the scrolling events are cached, but on rebuild they won't be)